errors/errors.go

```diff
@@ -47,19 +47,27 @@ func (p *Error) StackString() string {
 
 // VerboseString returns the error message, stack trace and contexts
 func (p *Error) VerboseString() string {
-	return fmt.Sprintf("%s\nPublic context: %+v\nPrivate context: %+v\n%s", p.Error(), p.PublicContext, p.PrivateContext, p.StackString())
+	return fmt.Sprintf("%s. %s\nParams: %+v\n%s", p.Error(), p.Message, p.Params, p.StackString())
+}
+
+func (p *Error) Format(f fmt.State, c rune) {
+	f.Write([]byte(p.Message))
 }
 
 // New creates a new error for you. Use this if you want to pass along a custom error code.
 // Otherwise use the handy shorthand factories below
-func New(code int, message string, context ...map[string]string) *Error {
-	return errorFactory(code, message, context...)
+func New(code string, message string, params map[string]string) *Error {
+	return errorFactory(code, message, params)
 }
 
 // Wrap takes any error interface and wraps it into an Error.
 // This is useful because an Error contains lots of useful goodies, like the stacktrace of the error.
 // NOTE: If `err` is already an `Error` the passed contexts will be ignored
```

Comment:
mentions `context` but should be `params`.

Alternatively, could we merge these into the existing error in the case the map is populated?